R/eatModel.r

```diff
@@ -1,55 +1,85 @@
 ### diverse functions that can be used on their own or are called by
-### multiple other functions
+### multiple other functions which couldn't be usefully grouped.
 
-### called by .substituteSigns() and getConquestAdditionalTerms () -------------
+### called by defineModel/.substituteSigns() and getResults/getConquestAdditionalTerms()
 
-isLetter <- function ( string ) {
+isLetter <- function(string){
+  # checks
+  checkmate::assert_character(string)

```

Comment:
I wasn't sure about what type "string" has, and it's difficult for me to find out what the exact goal of that function is or whether it needs checks at all. because it seems like the function returns lists with only the letters where anything else is removed.

Reply:
I would suggest to add assertions mainly for user-level functions. Internally used functions (like `isLetter()`) are usually called once the appropriateness of their input has already been checked at a higher level. I think assertion check should take place at the highest possible level.